checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,334 @@
+package org.checkerframework.checker.calledmethods;
+
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.Tree;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.builder.qual.ReturnsReceiver;
+import org.checkerframework.checker.calledmethods.framework.AutoValueSupport;
+import org.checkerframework.checker.calledmethods.framework.FrameworkSupport;
+import org.checkerframework.checker.calledmethods.framework.FrameworkSupportUtils;
+import org.checkerframework.checker.calledmethods.framework.LombokSupport;
+import org.checkerframework.checker.calledmethods.qual.CalledMethods;
+import org.checkerframework.checker.calledmethods.qual.CalledMethodsBottom;
+import org.checkerframework.checker.calledmethods.qual.CalledMethodsPredicate;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.common.accumulation.AccumulationAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.value.ValueAnnotatedTypeFactory;
+import org.checkerframework.common.value.ValueChecker;
+import org.checkerframework.common.value.ValueCheckerUtils;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+/** The annotated type factory for the Called Methods checker. */
+public class CalledMethodsAnnotatedTypeFactory extends AccumulationAnnotatedTypeFactory {
+
+    /** {@link java.util.Collections#singletonList} is treated specially by the EC2 logic. */
+    private final ExecutableElement collectionsSingletonList;
+
+    /**
+     * Whether to use the Value Checker as a subchecker to reduce false positives when analyzing
+     * calls to the AWS SDK. Defaults to false. Controlled by the command-line option {@code
+     * -AuseValueChecker}.
```

Comment:
Do we need to rename this option to mention it's for this checker?

Reply:
Good catch, we definitely should. I'm going to change it to `-AcalledMethodsUseValueChecker`, but I'm open to other names.

Do you think we should also rename `-AdisableFrameworkSupports` in a similar way?

Reply:
There is a general mechanism to pass an option only to a particular checker: https://github.com/typetools/checker-framework/blob/master/docs/manual/introduction.tex#L572
So including the checker name in the option seems like overkill to me.

Reply:
Thanks for pointing this out, @wmdietl. I've made the change.